### DISCUSSION TITLE
Error with Carla build at the step "make PythonAPI"

### DISCUSSION THREAD

**Discussion by vincentarnold448** [carla-simulator/carla#8392] [Q&A]
Hello, I tried to build Carla from source on Linux and have a problem at the command "make PythonAPI". When I try this, there comes much text,  but then the error which I wrote at the bottom. I don't know why this comes, I have installed clang 10.0. Does someone know how I can solve this problem? Thanks a lot for your answers. 

This was the error:

Elapsed Time Extracting boost for Python: 2 seconds
Building B2 engine..

A C++11 capable compiler is required for building the B2 engine.
Toolset 'clang' does not appear to support C++11.

> clang++ -x c++ -std=c++11  check_cxx11.cpp
check_cxx11.cpp:14:10: fatal error: 'thread' file not found
#include <thread>
         ^~~~~~~~
1 error generated.

** Note, the C++11 capable compiler is _only_ required for building the B2
** engine. The B2 build system allows for using any C++ level and any other
** supported language and resource in your projects.

You can specify the toolset as the argument, i.e.:
    ./build.sh [options] gcc

Toolsets supported by this script are:
    acc, clang, como, gcc, intel-darwin, intel-linux, kcc, kylix, mipspro,
    pathscale, pgi, qcc, sun, sunpro, tru64cxx, vacpp

For any toolset you can override the path to the compiler with the '--cxx'
option. You can also use additional flags for the compiler with the
'--cxxflags' option.

A special toolset; cxx, is available which is used as a fallback when a more
specific toolset is not found and the cxx command is detected. The 'cxx'
toolset will use the '--cxx' and '--cxxflags' options, if present.

Options:
    --help                  Show this help message.
    --verbose               Show messages about what this script is doing.
    --debug                 Build b2 with debug information, and no
                            optimizations.
    --guess-toolset         Print the toolset we can detect for building.
    --cxx=CXX               The compiler exec to use instead of the detected
                            compiler exec.
    --cxxflags=CXXFLAGS     The compiler flags to use in addition to the
                            flags for the detected compiler.

Failed to build B2 build engine
make: *** [Util/BuildTools/Linux.mk:142: setup] Error 1

**Comment by Yiran-SUSTech** [comment]
I also encountered the same problem, here is what I did to solve it, but I'm not sure if it will work in your case.

First, I use Ubuntu22.04, and I found it is probably the problem with Clang++, so I updated it to Clang++-12

`sudo apt update`
`sudo apt install clang-12 clang-tools-12 libc++-12-dev libc++abi-12-dev`
`sudo update-alternatives --config clang++`

and I manually add Clang 12 to `update-alternatives`:
`sudo update-alternatives --install /usr/bin/clang clang /usr/bin/clang-12 200`
`sudo update-alternatives --install /usr/bin/clang++ clang++ /usr/bin/clang++-12 200`

and choose it as the default version:
`sudo update-alternatives --config clang`
`sudo update-alternatives --config clang++`

then, I install libc++ and libstdc++, which have something to do with the header file "thread":
`sudo apt install libc++-12-dev libc++abi-12-dev libstdc++-12-dev`

and you can see if it works by running `make PythonAPI` again.
